### PR TITLE
Add nitro proxies

### PR DESCRIPTION
**Pull request by NullHypothesis**
### Type of change ( select one )

- [X] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: nitro-shim/tools/kafkaproxy/go.mod [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module github.com/brave-experiments/ia2-parent/kafkaproxy
+
+go 1.17
```
Comment (husobee) [inline nitro-shim/tools/kafkaproxy/go.mod:3]: can we bump to 1.18?

File: nitro-shim/tools/socksproxy/go.mod [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module github.com/brave-experiments/go-socks-proxy/v2
+
+go 1.17
```
Comment (husobee) [inline nitro-shim/tools/socksproxy/go.mod:3]: bump to 1.18

File: nitro-shim/tools/utils/go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module nitro-shim/utils
+
+go 1.17
```
Comment (husobee) [inline nitro-shim/tools/utils/go.mod:3]: bump 1.18

File: nitro-shim/tools/viproxy/go.mod [hunk 4]
```diff
@@ -0,0 +1,18 @@
+module github.com/brave-experiments/ia2-parent/viproxy
+
+go 1.17
```
Comment (husobee) [inline nitro-shim/tools/viproxy/go.mod:3]: bump 1.18

**Review by husobee — APPROVED** [review]

File: nitro-shim/tools/kafkaproxy/go.mod (hunk 1 above)
Comment (NullHypothesis) [reply nitro-shim/tools/kafkaproxy/go.mod]: Should be addressed in 4e928e5.

File: nitro-shim/tools/utils/go.mod (hunk 3 above)
Comment (NullHypothesis) [reply nitro-shim/tools/utils/go.mod]: Should be addressed in https://github.com/brave-intl/bat-go/commit/4e928e5b0d36c6f5e6b195ea73e79b07a86690a1.

File: nitro-shim/tools/viproxy/go.mod (hunk 4 above)
Comment (NullHypothesis) [reply nitro-shim/tools/viproxy/go.mod]: Should be addressed in https://github.com/brave-intl/bat-go/commit/4e928e5b0d36c6f5e6b195ea73e79b07a86690a1.

**Pull request merged**